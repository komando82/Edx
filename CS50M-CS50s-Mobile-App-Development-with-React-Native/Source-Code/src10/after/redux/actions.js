import {login} from '../api'

// action types
export const UPDATE_USER = 'UPDATE_USER'
export const UPDATE_CONTACT = 'UPDATE_CONTACT'
export const LOG_IN_SENT = 'LOG_IN_SENT'
export const LOG_IN_FULFILLED = 'LOG_IN_FULFILLED'
export const LOG_IN_REJECTED = 'LOG_IN_REJECTED'

// action creators
export const updateUser = update => ({
  type: UPDATE_USER,
  payload: update,
})

export const addContact = newContact => ({
  type: UPDATE_CONTACT,
  payload: newContact,
})

// async action creator
export const logInUser = (username, password) => async dispatch => {
  dispatch({type: LOG_IN_SENT})
  try {
    const token = await login(username, password)
    dispatch({type: LOG_IN_FULFILLED, payload: token})
  } catch (err) {
    dispatch({type: LOG_IN_REJECTED, payload: err.message})
  }
}
