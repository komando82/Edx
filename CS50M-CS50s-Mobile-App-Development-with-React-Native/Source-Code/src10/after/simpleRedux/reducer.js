const merge = (prev, next) => Object.assign({}, prev, next)

const reducer = (state, update) => merge(state, update)

let state = {}
state = reducer(state, {foo: 'foo'})
state = reducer(state, {bar: 'bar'})
state = reducer(state, {foo: 'baz'})

console.log(state)
