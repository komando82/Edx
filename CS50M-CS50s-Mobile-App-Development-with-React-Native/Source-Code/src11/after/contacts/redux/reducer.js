import {combineReducers} from 'redux'

import contacts from '../contacts'

import {CHANGE_FIRST_CONTACT, UPDATE_USER, UPDATE_CONTACT, LOG_IN_FULFILLED, LOG_IN_REJECTED} from './actions'

const merge = (prev, next) => Object.assign({}, prev, next)

const contactReducer = (state = contacts, action) => {
  if (action.type === UPDATE_CONTACT) return [...state, action.payload]
  if (action.type === CHANGE_FIRST_CONTACT) {
    const [firstContact, ...rest] = state
    if (!firstContact) return state
    const newContact = {...firstContact, name: 'Jordan Hayashi'}
    return [newContact, ...rest]
  }
  return state
}

const userReducer = (state = {}, action) => {
  switch (action.type) {
    case UPDATE_USER:
      return merge(state, action.payload)
    case UPDATE_CONTACT:
      return merge(state, {prevContact: action.payload})
    case LOG_IN_FULFILLED:
      return merge(state, {token: action.payload})
    case LOG_IN_REJECTED:
      return merge(state, {loginErr: action.payload})
    default:
      return state
  }
}

const reducer = combineReducers({
  user: userReducer,
  contacts: contactReducer,
})

export default reducer
