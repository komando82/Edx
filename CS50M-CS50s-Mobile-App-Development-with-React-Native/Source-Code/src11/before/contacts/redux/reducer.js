import {combineReducers} from 'redux'

import contacts from '../contacts'

import {UPDATE_USER, UPDATE_CONTACT, LOG_IN_FULFILLED, LOG_IN_REJECTED} from './actions'

const merge = (prev, next) => Object.assign({}, prev, next)

const contactReducer = (state = contacts, action) => {
  if (action.type === UPDATE_CONTACT) return [...state, action.payload]
  return state
}

const userReducer = (state = {}, action) => {
  switch (action.type) {
    case UPDATE_USER:
      return merge(state, action.payload)
    case UPDATE_CONTACT:
      return merge(state, {prevContact: action.payload})
    case LOG_IN_FULFILLED:
      return merge(state, {token: action.payload})
    case LOG_IN_REJECTED:
      return merge(state, {loginErr: action.payload})
    default:
      return state
  }
}

const reducer = combineReducers({
  user: userReducer,
  contacts: contactReducer,
})

export default reducer
