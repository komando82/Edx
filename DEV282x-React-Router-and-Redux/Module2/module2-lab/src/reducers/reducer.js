import { combineReducers } from "redux";

import selectedTable from "./selectedTable.js";
import tableStatusData from "./tableStatusData.js";
import tableData from "./tableData.js";
import moneyEarned from "./moneyEarned.js";

const reducer = combineReducers({
  selectedTable,
  tableStatusData,
  tableData,
  moneyEarned
});

export default reducer;