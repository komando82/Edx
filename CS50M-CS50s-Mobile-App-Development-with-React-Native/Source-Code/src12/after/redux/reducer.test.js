/* eslint-disable */

import reducer from './reducer'
import * as actions from './actions'

const DEFAULT_STATE = {
  user: {},
  contacts: [],
}

describe('contact reducer', () => {
  it('successfully adds new user', () => {
    expect(reducer(DEFAULT_STATE, actions.addContact({
      name: 'test user',
      phone: '1234567890',
    }))).toMatchSnapshot()
  })
})

describe('user reducer', () => {
  it('successfully updates user', () => {
    expect(reducer(DEFAULT_STATE, actions.updateUser({
      name: 'test user',
    }))).toMatchSnapshot()
  })
})
