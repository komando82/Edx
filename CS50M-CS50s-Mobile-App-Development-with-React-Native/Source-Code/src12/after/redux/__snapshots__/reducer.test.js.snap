// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact reducer successfully adds new user 1`] = `
Object {
  "contacts": Array [
    Object {
      "name": "test user",
      "phone": "1234567890",
    },
  ],
  "user": Object {
    "prevContact": Object {
      "name": "test user",
      "phone": "1234567890",
    },
  },
}
`;

exports[`user reducer successfully updates user 1`] = `
Object {
  "contacts": Array [],
  "user": Object {
    "name": "test user",
  },
}
`;
